pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -971,6 +1005,8 @@ default 'raise'
                                                  dts.hour, dts.min, dts.sec,
                                                  dts.us),
                                         is_dst=not bool(fold))
+            # set fold after pytz resets it to 0 (GH 25057)
+            ts_input = ts_input.replace(fold=fold)
```

Comment:
This is `pytz`-specific fold logic, right? We can drop it?

Reply:
Thanks, missed this.